subprojects/docs/src/docs/userguide/dep-man/06-publishing/publishing.adoc

```diff
@@ -300,13 +300,16 @@ include::sample[dir="userguide/dependencyManagement/modelingFeatures/crossProjec
 
 [WARNING]
 ====
-When you're attaching extra artifacts to a publication, remember that they are _secondary_ artifacts that support a _primary_ artifact. The metadata that a publication defines — such as dependency information — is associated with that primary artifact only. Thinking about publications in this way should help you determine whether you should be adding custom artifacts to an existing publication, or defining a new publication.
-
-To some extent, you can see those extra artifacts as being published "out of context".
-The only way to address them as a consumer is to either to download them directly or use a _classifier_ notation.
+Instead of thinking in terms of artifacts, you should really embrace the variant aware model of Gradle.
+It is expected that a single module may need multiple artifacts.
+However this rarely stops there, if the additional artifacts represent an <<feature_variants.adoc#,optional feature>>, they might also have different dependencies and more.
 
 Gradle, via _Gradle Module Metadata_, supports the publication of _additional variants_ which make those artifacts known to the dependency resolution engine.
 Please refer to the <<modeling_features.adoc#sec:variant-aware-sharing, variant-aware sharing>> section of the documentation to see how to declare such variants and <<#sec:publishing-custom-components, check out how to publish custom components>>.
+
+To some extent, you can see those extra artifacts as being published "out of context".
+If you attach extra artifacts to a publication, remember that they are _secondary_ artifacts only.
+The Maven metadata will only allow addressing these artifacts through the `classifier`.
```

Comment:
Reorder sentences for clarification (?):

If you attach extra artifacts to a publication directly, they are published "out of context".
That means, they are not referenced in the metadata at all and can then only be addressed directly through a _classifier_ on a dependency.
Note that, in contrast to Gradle Module Metadata, Maven pom metadata will not contain information on additional artifacts regardless of whether they are added though a variant or directly, as variants can not be represented in the pom format.